alpha/apps/kaltura/lib/kPermissionManager.php

```diff
@@ -1120,18 +1120,23 @@ protected static function handleRecycleBinPermission($permission)
 		if ($permission->getStatus() == PermissionStatus::ACTIVE)
 		{
 			self::enableRecycleBinScheduledTaskProfile($permission->getPartnerId());
-			$scheduledTaskPluginPermission = PermissionPeer::getByNameAndPartner('SCHEDULEDTASK_PLUGIN_PERMISSION', array($permission->getPartnerId()));
-			if ($scheduledTaskPluginPermission->getStatus() != PermissionStatus::ACTIVE)
-			{
-				$scheduledTaskPluginPermission->setStatus(PermissionStatus::ACTIVE);
-				$scheduledTaskPluginPermission->save();
-			}
+			self::enableRequiredPluginsPermissions($permission->getPartnerId(), PermissionName::SCHEDULEDTASK_PLUGIN_PERMISSION);
 		}
 		elseif ($permission->getStatus() == PermissionStatus::BLOCKED)
 		{
 			self::disableRecycleBinScheduledTaskProfile($permission->getPartnerId());
 		}
 	}
+
+	protected static function enableRequiredPluginsPermissions($partnerId, $permissionName)
+	{
+		$scheduledTaskPluginPermission = PermissionPeer::getByNameAndPartner($permissionName, array($partnerId));
```

Comment:
if you allredy made the function generic why call it explicitly scheduledTaskPluginPermission?